padertorch/contrib/examples/tasnet/train.py

```diff
@@ -0,0 +1,309 @@
+    load_model_from = None
+    database_json = None
+
+    if database_json is None:
+        raise MissingConfigError(
+            'You have to set the path to the database JSON!', 'database_json')
+    if not Path(database_json).exists():
+        raise InvalidConfigError('The database JSON does not exist!',
+                                 'database_json')
+
+    # Start with an empty dict to allow tracking by Sacred
+    trainer = {
+        "model": {
+            "factory": 'padertorch.contrib.examples.tasnet.tasnet.TasNet',
+        },
+        "storage_dir": None,
+        "optimizer": {
+            "factory": pt.optimizer.Adam,
+            "gradient_clipping": 1
+        },
+        "summary_trigger": (1000, "iteration"),
+        "stop_trigger": (100_000, "iteration"),
+        "loss_weights": {
+            "si-sdr": 1.0,
+            "log-mse": 0.0,
+            "si-sdr-grad-stop": 0.0,
+        }
+    }
+    pt.Trainer.get_config(trainer)
+    if trainer['storage_dir'] is None:
+        trainer['storage_dir'] = get_storage_dir()
+
+    ex.observers.append(FileStorageObserver(
+        Path(trainer['storage_dir']) / 'sacred')
+    )
+
+
+@ex.named_config
+def win2():
+    """
+    This is the configuration for the best performing model from the DPRNN
+    paper. Training takes very long time with this configuration.
+    """
+    # The model becomes very memory consuming with this small window size.
+    # You might have to reduce the chunk size as well.
+    batch_size = 1
+
+    trainer = {
+        'model': {
+            'encoder_block_size': 2,
+            'dprnn_window_length': 250,
+            'dprnn_hop_size': 125,  # Half of window length
+        }
+    }
+
+
+@ex.named_config
+def log_mse():
+    trainer = {
+        'loss_weights': {
+            'si-sdr': 0.0,
+            'log-mse': 1.0,
+        }
+    }
+
+
+@ex.named_config
+def on_wsj0_2mix_max():
+    chunk_size = -1
+    train_dataset = "mix_2_spk_max_tr"
+    validate_dataset = "mix_2_spk_max_cv"
+
+
+@ex.capture
+def pre_batch_transform(inputs):
+    return {
+        's': np.ascontiguousarray([
+            load_audio(p)
+            for p in inputs['audio_path']['speech_source']
+        ], np.float32),
+        'y': np.ascontiguousarray(
+            load_audio(inputs['audio_path']['observation']), np.float32),
+        'num_samples': inputs['num_samples'],
+        'example_id': inputs['example_id'],
+        'audio_path': inputs['audio_path'],
+    }
+
+
+def prepare_iterable(
+        db, dataset: str, batch_size, chunk_size, prefetch=True,
+        iterator_slice=None
+):
+    """
+    This is re-used in the evaluate script
+    """
+    iterator = db.get_dataset(dataset)
+
+    if iterator_slice is not None:
+        iterator = iterator[iterator_slice]
+
+    iterator = (
+        iterator
+        .map(pre_batch_transform)
+        .map(RandomChunkSingle(chunk_size, chunk_keys=('y', 's'), axis=-1))
+        .shuffle(reshuffle=False)
```

Comment:
Is there a reason, why you disabled reshuffle? Here it should be supported.

Reply:
No, there's no reason for that. I think this was copied from the pit example

Reply:
Can you change this to `reshuffle=True`? The PIT example was written, before https://github.com/fgnt/lazy_dataset/pull/2 was merged.

Reply:
Done!